providers/mlx4/qp.c

```diff
@@ -38,11 +38,12 @@
 #include <pthread.h>
 #include <string.h>
 #include <errno.h>
+#include <util/mmio.h>
 #include <util/compiler.h>
 
 #include "mlx4.h"
-#include "doorbell.h"
 #include "wqe.h"
+#include "mmio.h"
```

Comment:
No, at this particular point in the patch series the special mlx4 mmio.h still has required content, both are required to be included. At the memcpy patch mlx4/mmio.h is removed.